### DISCUSSION TITLE
Calculation with d_vectors

### DISCUSSION THREAD

**Discussion by tieincred** [coqui-ai/TTS#1665] [General Q&A] [answered]
Hello,
Please forgive for naïve question as I am need to the tech.
I am trying to train a multi speaker model on vits but I am unable to understand do we have to calculate the d_vectors for speakers using the TTS/bin/compute_embeddings.py file?
Or using the recipe it'll be calculated automatically and saved because there is nothing mentioned as such in recipe or am I missing something?
Secondly about characters, I am training on a different language than english but at many discussions I've seen people either specifying characters and sometimes not.
Can you please guide me how important it is to specify characters set?

**Comment by p0p4k** [comment, marked as answer]
Hello.
1) You need to calculate the d-vectors for the audio files using compute_embeddings.py file. Use the speaker_encoder_models from the assets of the latest/corresponding release. The files generated are then referred to in the train.py configs.
2) AI model does not read the letters (characters). Instead it reads "tokens". English basic tokens are the alphabets (a,b,c, etc.). But some other languages (like Hindi, Korean, etc.) can have consonants+vowels as base tokens or can be broken further into individual consonants and vowels separately. Hence your character set is based on your language.

**Reply by tieincred** [reply]
@p0p4k , Thanks a lot for the reply.
about the second point, I understand about different character set for different language, but I have seen people training on a language other than english without specifying characters, the training still happens.
So, I am confused as how important it is to specify characters.

**Reply by p0p4k** [reply]
The best answer for your case is try both the ways - with and without characters specification and see what works best for you :) Good luck!

**Reply by tieincred** [reply]
Thank you for answers

**Reply by p0p4k** [reply]
> Thank you for answers

No problem. Keep us updated about your training progress, as well.
